crmsh/corosync.py

```diff
@@ -386,6 +386,11 @@ def add_node_ucast(IParray, node_name=None):
     f = open(conf()).read()
     p = Parser(f)
 
+    exist_iplist = p.get_all('nodelist.node.ring0_addr')
+    for ip in IParray:
+        if ip in exist_iplist:
+            return
+
```

Comment:
Ah, this doesn't look quite right: The IParray contains [ring0_addr, ring1_addr, ..., ringN_addr], right?
Really, this should get all configured IP addresses, and if any of the input IP addresses is already in use, raise an exception.

So I'd write a function `get_all_configured_addresses` which returns a list of all configured addresses, and then check if any of the input addresses is in that list. If so, `raise ValueError("IP {} was already configured")` - and then make sure to catch and handle that.

Reply:
Thank you!

Reply:
I have changed the logic to make sure extract all configured IP.

And I think if this case does happen, we should better keep silence and complete the remain bootstrap process to make sure join node will start cluster successfully, 
So just return, no raise error

Reply:
What you can do is raise the error here (since that is the correct response in the general case), and catch the error at the usage site to ignore it there.

Reply:
OK, changed